app/controllers/users/profiles_controller.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module Users
+  class ProfilesController < ApplicationController
+    before_action :set_profile
+    before_action :authenticate_user!
+
+    def index
+      if user_signed_in?
```

Comment:
1. add https://github.com/varvet/pundit to the project
2. get familiar with it, introduce policies to the project
3. `authorize index?`
4. catch exception in ApplicationController, generically redirect to sign_in page